lib/logstash/inputs/http.rb

```diff
@@ -0,0 +1,100 @@
+# encoding: utf-8
+require "logstash/inputs/base"
+require "logstash/namespace"
+require "stud/interval"
+require "socket" # for Socket.gethostname
+require "puma/server"
+require "puma/minissl"
+
+class Puma::Server
+  # ensure this method doesn't mess up our vanilla request
+  def normalize_env(env, client); end
+end
+
+# This output opens a web server and listens to HTTP requests,
+# converting them to LogStash::Event instances.
+#
+class LogStash::Inputs::Http < LogStash::Inputs::Base
+  config_name "http"
+
+  # If undefined, Logstash will complain, even if codec is unused.
+  default :codec, "plain"
+
+  # Which host or ip to bind to
+  config :host, :validate => :string, :default => "0.0.0.0"
+
+  # Which TCP port to bind to
+  config :port, :validate => :number, :default => 8080
+
+  # Maximum number of threads to use
+  config :max_threads, :validate => :number, :default => 4
```

Comment:
but since this is a Puma-specific setting, maybe I should change this to a more puma-specific name like puma_max_threads?